src/Akka.Cluster.Hosting/AkkaClusterHostingExtensions.cs

```diff
@@ -233,6 +233,12 @@ public sealed class ShardOptions
         /// The interval between retries for acquiring the lease
         /// </summary>
         public TimeSpan? LeaseRetryInterval { get; set; }
+        
+        /// <summary>
+        /// The message that will be sent to entities when they are to be stopped for a rebalance or
+        /// graceful shutdown of a <see cref="Sharding.ShardRegion"/>, e.g. <see cref="PoisonPill"/>.
+        /// </summary>
+        public object? HandOffStopMessage { get; set; }
```

Comment:
Hand off stop message - now exposed via the `ShardOptions` class so it can be customized by end-users.